libs/features/icons/icons.js

```diff
@@ -36,39 +39,74 @@ export const fetchIcons = (config) => new Promise(async (resolve) => {
-    const existingIcon = icon.querySelector('svg');
-    if (!iconSVGs[iconName] || existingIcon) return;
+export function setNodeIndexClass(icon) {
+  const parent = icon.parentNode;
+  const children = parent.childNodes;
+  const nodeIndex = Array.prototype.indexOf.call(children, icon);
+  let indexClass = (nodeIndex === children.length - 1) ? 'last' : 'middle';
+  if (nodeIndex === 0) indexClass = 'first';
+  if (children.length === 1) indexClass = 'only';
+  icon.classList.add(`node-index-${indexClass}`);
+}
+
+export default async function loadIcons(icons) {
+  const fedRoot = getFederatedContentRoot();
+  const iconRequests = [];
+  const iconsToFetch = new Map();
+
+  icons.forEach((icon) => {
+    setNodeIndexClass(icon);
+    if (icon.classList.contains('icon-tooltip')) decorateToolTip(icon);
+    const iconName = [...icon.classList].find((c) => c.startsWith('icon-'))?.substring(5);
+    if (icon.dataset.svgInjected || !iconName) return;
+    if (!federalIcons[iconName] && !iconsToFetch.has(iconName)) {
+      const url = `${fedRoot}/federal/assets/icons/svgs/${iconName}.svg`;
+      iconsToFetch.set(iconName, fetch(url)
+        .then(async (res) => {
+          if (!res.ok) throw new Error(`Failed to fetch SVG for ${iconName}: ${res.statusText}`);
+          const text = await res.text();
+          const parser = new DOMParser();
+          const svgDoc = parser.parseFromString(text, 'image/svg+xml');
+          const svgElement = svgDoc.querySelector('svg');
+          if (!svgElement) {
+            window.lana?.log(`No SVG element found in fetched content for ${iconName}`);
+            return;
+          }
+          const svgClone = svgElement.cloneNode(true);
+          svgClone.classList.add('icon-milo', `icon-milo-${iconName}`);
+          federalIcons[iconName] = svgClone;
+        })
+        /* c8 ignore next 3 */
+        .catch((error) => {
+          window.lana?.log(`Error fetching SVG for ${iconName}:`, error);
+        }));
+    }
+    iconRequests.push(iconsToFetch.get(iconName));
     const parent = icon.parentElement;
-    if (parent.childNodes.length > 1) {
-      if (parent.lastChild === icon) {
-        icon.classList.add('margin-inline-start');
-      } else if (parent.firstChild === icon) {
-        icon.classList.add('margin-inline-end');
-        if (parent.parentElement.tagName === 'LI') parent.parentElement.classList.add('icon-list-item');
-      } else {
-        icon.classList.add('margin-inline-start', 'margin-inline-end');
-      }
+    if (parent && parent.parentElement.tagName === 'LI') parent.parentElement.classList.add('icon-list-item');
+  });
+
+  await Promise.all(iconRequests);
+
+  icons.forEach((icon) => {
+    const iconName = [...icon.classList].find((c) => c.startsWith('icon-'))?.substring(5);
+    if (iconName && federalIcons[iconName] && !icon.dataset.svgInjected) {
+      const svgClone = federalIcons[iconName].cloneNode(true);
```

Comment:
Why do we need to clone the SVG yet again, couldn't we reference it directly at this point?

Reply:
I think we need this by I may be wrong.. I tried to ref it directly and duplicate icons on a page don't get decorated w/ out the cloned node